source/extensions/access_loggers/http_grpc/grpc_access_log_impl.cc

```diff
@@ -274,15 +273,43 @@ void HttpGrpcAccessLog::log(const Http::HeaderMap* request_headers,
   }
   request_properties->set_request_headers_bytes(request_headers->byteSize());
   request_properties->set_request_body_bytes(request_info.bytesReceived());
+  if (request_headers->Method() != nullptr) {
+    envoy::api::v2::core::RequestMethod method =
+        envoy::api::v2::core::RequestMethod::METHOD_UNSPECIFIED;
+    envoy::api::v2::core::RequestMethod_Parse(
+        std::string(request_headers->Method()->value().c_str()), &method);
+    request_properties->set_request_method(method);
+  }
+  if (config_.additional_request_headers_to_log_size() > 0) {
+    auto* logged_headers = request_properties->mutable_request_headers();
+
+    for (const auto& additional_header : config_.additional_request_headers_to_log()) {
+      const Http::HeaderEntry* entry =
+          request_headers->get(Http::LowerCaseString(additional_header));
```

Comment:
perf nit: Can we pre-convert the config to LowerCaseString here and below to avoid the temporary for each log?

Reply:
I did it this way so that the headers would be logged with the requested-case, but recording them always-as-lowercase seems fine as well.

Reply:
Done.